win32ss/user/ntuser/desktop.c

```diff
@@ -1788,6 +1788,9 @@ BOOL IntDeRegisterShellHookWindow(HWND hWnd)
     PLIST_ENTRY ListEntry;
     PSHELL_HOOK_WINDOW Current;
 
+    if (Desktop == NULL)
```

Comment:
In which situation did you hit that Desktop == NULL case?

Reply:
It happens during user32_winetest msg_focus:
```
nt!KiPageFault+0x150
win32k!IntDeRegisterShellHookWindow+0x28 [E:\amd64\reactos\win32ss\user\ntuser\desktop.c @ 1791] 
win32k!co_UserFreeWindow+0x1c9 [E:\amd64\reactos\win32ss\user\ntuser\window.c @ 603] 
win32k!co_UserDestroyWindow+0x8a5 [E:\amd64\reactos\win32ss\user\ntuser\window.c @ 3006] 
win32k!UserDestroyObjectsForOwner+0x9e [E:\amd64\reactos\win32ss\user\ntuser\object.c @ 773] 
win32k!ExitThreadCallback+0x3e0 [E:\amd64\reactos\win32ss\user\ntuser\main.c @ 782] 
win32k!Win32kThreadCallback+0xe5 [E:\amd64\reactos\win32ss\user\ntuser\main.c @ 898] 
nt!PspExitThread+0xb67 [E:\amd64\reactos\ntoskrnl\ps\kill.c @ 751] 
```
For some reason the thread doesn't have a desktop at this point. I do not know why, but I know it's better to not crash the system.

Reply:
Interesting. This could only happen for the desktop thread as I understand. The desktop thread owns all desktop windows for all desktops, and the desktop windows are allocated from the desktop heap. The real question then is why this hasn't happened before.